api/dispatcher.proto

```diff
@@ -170,6 +170,23 @@ message AssignmentsRequest {
 	string session_id = 1 [(gogoproto.customname) = "SessionID"];
 }
 
+message Assignment {
+	oneof item {
+		Task task = 2;
+		Secret secret = 3;
```

Comment:
We do this to avoid creating another enum type for each message type. The packed object works similar to a tagged enum, without extra pageantry.